grr/server/grr_response_server/gui/ui/store/client_page_facade.ts

```diff
@@ -157,10 +157,18 @@ export class ClientPageStore extends ComponentStore<ClientPageState> {
 
   /** An observable emitting the client loaded by `selectClient`. */
   readonly selectedClient$: Observable<Client> =
-      this.select(state => state.client)
+      combineLatest(
+        interval(this.configService.config.selectedClientPollingIntervalMs)
           .pipe(
-              filter((client): client is Client => client !== undefined),
-          );
+            switchMapTo(this.select(state => state.client)),
+            filter((client): client is Client => client !== undefined),
```

Comment:
This `pipe()` should only trigger the loading of the Client data by calling `selectClient()`. Don't load the data here, this might be the reason why your tests are failing.

Reply:
I tried refactoring the method to better suit our needs, and added a clientId variable to the store